contributors/design-proposals/hugepages.md

```diff
@@ -0,0 +1,213 @@
+## Abstract
+
+A proposal to allow huge page use by applications running in a Kubernetes
+cluster.
+
+A pod should be able to have a number of huge pages for use by the
+application.  The scheduler should be able have visibility into the node
+capacity of huge pages, for each huge page size, and make a decision about if
+the pod can be scheduled on that node.  The kubelet should report the number of
+available huge pages and set up the environment such that the pod can
+successfully use the number of huge pages requested in the pod definition.
+
+## Motivation
+
+Huge page support is needed for many large memory HPC workloads to achieve
+acceptable performance levels.
+
+This proposal is part of a larger effort to better support High Performance
```

Comment:
HPC is too loaded of a term.  It's really just performance sensitive workloads.  JVMs with large heaps, stateful applications with large in-memory caches, even memcached, etc.